packages/manager/apps/telecom/src/app/telecom/telephony/service/contact/contact.controller.js

```diff
@@ -50,379 +162,297 @@ export default /* @ngInject */ function TelecomTelephonyServiceContactCtrl(
+      this.directoryForm.siret,
+    ).then((infos) => {
+      if (infos.status === 'todo') {
+        return this.$timeout(
+          () => this.fetchEntrepriseInformations(this.directoryForm.siret),
+          500,
         );
-      })
-      .catch((err) => new TucToastError(err))
-      .finally(() => {
-        self.isUpdating = false;
-      });
-  };
-
-  self.bulkDatas = {
-    billingAccount: $stateParams.billingAccount,
-    serviceName: $stateParams.serviceName,
-    infos: {
-      name: 'contact',
-      actions: [
-        {
-          name: 'directory',
-          route: '/telephony/{billingAccount}/service/{serviceName}/directory',
-          method: 'PUT',
-          params: null,
-        },
-      ],
-    },
-  };
-
-  self.getBulkParams = function getBulkParams() {
-    return pickBy(
-      self.directoryForm,
-      (value, key) => !get(self.directoryProperties, [key, 'readOnly']),
+      }
+      return infos;
+    });
+  }
+
+  postCodeAvailable(directory) {
+    return this.TelecomTelephonyServiceContactService.getPostCodeAvailable(
+      this.$stateParams.serviceName,
+      directory.country,
     );
-  };
+  }
 
-  self.onBulkSuccess = function onBulkSuccess(bulkResult) {
-    // display message of success or error
-    tucTelephonyBulk
-      .getTucToastInfos(bulkResult, {
-        fullSuccess: $translate.instant(
-          'telephony_service_contact_bulk_all_success',
-        ),
-        partialSuccess: $translate.instant(
-          'telephony_service_contact_bulk_some_success',
-          {
-            count: bulkResult.success.length,
-          },
-        ),
-        error: $translate.instant('telephony_service_contact_bulk_error'),
-      })
-      .forEach((toastInfo) => {
-        TucToast[toastInfo.type](toastInfo.message, {
-          hideAfter: null,
+  onPostCodeChange() {
+    this.directoryForm.city = '';
+    this.directoryForm.cedex = '';
+    this.directoryForm.wayNumber = '';
+    this.directoryForm.wayNumberExtra = '';
+    this.directoryForm.wayName = '';
+    this.directoryForm.addressExtra = '';
+
+    if (this.directoryForm.postCode !== this.directory.postCode) {
+      this.directoryForm.urbanDistrict =
+        this.isUrbanDistrictRequired() &&
+        this.directoryForm.postCode.length === 5
+          ? this.directoryForm.postCode.substring(3, 5)
+          : '';
+    }
+
+    // Fetch cities for given post code
+    if (this.directoryForm.postCode?.length >= 3) {
+      this.$q
+        .resolve(
+          this.TelecomTelephonyServiceContactService.getCityAvailable(
+            this.directoryForm.postCode,
+            this.directory.country,
+          ),
+        )
+        .then((cities) => {
+          this.autoCompleteCity =
+            cities?.length > 1
+              ? cities.sort((a, b) => {
+                  if (a < b) return -1;
+                  if (a > b) return 1;
+                  return 0;
+                })
```

Comment:
**suggestion**: Sorting function can be moved to different helper function instead of repeating.

Reply:
I had the impression that it was used 3 times but in the end, one of the 3 is not necessary because the sorting is already done on the api side. And the 2nd one is slightly different a, b becomes a.name and b.name in the function.

Reply:
I succeeded in needing it only once.